hicommonwealth/commonwealth
# [Launchpad Discovery]: Randomize token idea generation

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8863

## Description of Changes
- Added randomize button functionality with content streaming for a better UX experience
- Added typing animation to fill the form as the idea chunks are generated
- Added draft states to handle multiple generated ideas
- Added optional initial prompt text field to generate an idea

## "How We Fixed It"
N/A

## Test Plan


1. Enable `FLAG_TOKENIZED_COMMUNITY`
2. Visit the `/communities` page
3. Click on the "Launch token" button in the "launch an idea" card verify the drawer opens with the token launch form.
4. Click on the randomize button, verify you see the form getting filled with a token idea with a typing animation
5. Verify you can generate up to 5 ideas with the randomize button
6. Verify you can switch b/w the generated ideas using the idea counter
<img width="183" alt="image" src="https://github.com/user-attachments/assets/da3d0944-849a-4449-a009-446d4b3f6c35">

7. Verify you are able to launch a generated idea successfully
8. Now visit the `/communities` page again
9. This time click on the randomize button directly, and verify you see the token drawer open with the form getting filled with the typing animation.
10. Now close the drawer again
11. Add some idea prompt in the text field of the launch token card, and click on the randomize button again
12. Verify this time the generated idea matches your prompt
13. Verify you are able to generate multiple ideas simultaneously
14. Verify you can switch b/w multiple ideas as they are being generated
15. Verify you can't launch the actively generated idea using the "Token launch" button while it is being generated (i.e when form fields are disabled)
16. Verify you can't edit the form while the active idea is being generated
17. Verify you are able to update the draft state/values of the generated idea once it's finished generating.


## Deployment Plan
N/A

## Other Considerations
N/A
